redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java

```diff
@@ -6,590 +6,24 @@
-import java.util.HashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-import java.util.TreeMap;
-
-import org.hibernate.ogm.datastore.document.impl.DotPatternMapHelpers;
-import org.hibernate.ogm.datastore.document.options.AssociationStorageType;
-import org.hibernate.ogm.datastore.document.options.spi.AssociationStorageOption;
-import org.hibernate.ogm.datastore.map.impl.MapHelpers;
-import org.hibernate.ogm.datastore.redis.dialect.model.impl.RedisAssociation;
-import org.hibernate.ogm.datastore.redis.dialect.model.impl.RedisAssociationSnapshot;
-import org.hibernate.ogm.datastore.redis.dialect.model.impl.RedisTupleSnapshot;
 import org.hibernate.ogm.datastore.redis.dialect.value.Association;
 import org.hibernate.ogm.datastore.redis.dialect.value.Entity;
 import org.hibernate.ogm.datastore.redis.impl.RedisDatastoreProvider;
-import org.hibernate.ogm.datastore.redis.impl.json.JsonEntityStorageStrategy;
-import org.hibernate.ogm.datastore.redis.impl.json.JsonSerializationStrategy;
-import org.hibernate.ogm.datastore.redis.options.impl.TTLOption;
-import org.hibernate.ogm.dialect.multiget.spi.MultigetGridDialect;
-import org.hibernate.ogm.dialect.spi.AssociationContext;
-import org.hibernate.ogm.dialect.spi.AssociationTypeContext;
-import org.hibernate.ogm.dialect.spi.BaseGridDialect;
-import org.hibernate.ogm.dialect.spi.ModelConsumer;
-import org.hibernate.ogm.dialect.spi.NextValueRequest;
-import org.hibernate.ogm.dialect.spi.TupleContext;
-import org.hibernate.ogm.model.key.spi.AssociationKey;
-import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
-import org.hibernate.ogm.model.key.spi.AssociationKind;
-import org.hibernate.ogm.model.key.spi.AssociationType;
-import org.hibernate.ogm.model.key.spi.EntityKey;
-import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
-import org.hibernate.ogm.model.key.spi.IdSourceKey;
-import org.hibernate.ogm.model.key.spi.RowKey;
-import org.hibernate.ogm.model.spi.Tuple;
-import org.hibernate.ogm.model.spi.TupleOperation;
-import org.hibernate.ogm.options.spi.OptionsContext;
-import org.hibernate.ogm.type.spi.GridType;
-import org.hibernate.type.Type;
-
-import com.lambdaworks.redis.KeyScanCursor;
-import com.lambdaworks.redis.RedisConnection;
-import com.lambdaworks.redis.ScanArgs;
-
-import static org.hibernate.ogm.datastore.document.impl.DotPatternMapHelpers.getColumnSharedPrefixOfAssociatedEntityLink;
 
 /**
  * Stores tuples and associations inside Redis.
- * <p>
+ * <p/>
  * Tuples are stored in Redis as a JSON serialization of a {@link Entity} object. Associations are stored in Redis obtained as a
  * JSON serialization of a {@link Association} object either within the entity or external.
  * See {@link org.hibernate.ogm.datastore.document.cfg.DocumentStoreProperties#ASSOCIATIONS_STORE} on how to configure
  * entity or external storage.
  *
  * @author Mark Paluch
+ * @deprecated Use {@link RedisJsonDialect}
```

Comment:
This could be just removed. We'd only need to deprecate it if had been part of a non-preview (alpha, beta etc.) release.